src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -1460,6 +1465,56 @@ public function configureProductPaginationAction(Request $request): JsonResponse
         return $this->json(null, Response::HTTP_NO_CONTENT);
     }
 
+    /**
+     * Method for downloading customization picture
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param int $orderId
+     * @param int $type
+     * @param string $name
+     * @param string $value
+     *
+     * @return BinaryFileResponse|RedirectResponse
+     */
+    public function displayCustomizationImageAction(int $orderId, int $type, string $name, string $value)
+    {
+        $uploadDir = $this->get('prestashop.adapter.legacy.context')->getUploadDirectory();
+        $customizationForViewing = new OrderProductCustomizationForViewing($type, $name, $value);
+        $file = $uploadDir . $customizationForViewing->getValue();
+        $mimeTypeGuesser = new FileinfoMimeTypeGuesser();
+        $filesystem = new Filesystem();
+
+        try {
+            if (!$filesystem->exists($file)) {
+                $this->addFlash('error', $this->trans('The product customization image file was not found.', 'Admin.Notifications.Error'));
+
+                return $this->redirectToRoute('admin_orders_view', [
+                    'orderId' => $orderId,
+                ]);
+            }
+            $response = new BinaryFileResponse($file);
+
+            $response->headers->set(
+                'Content-Type',
+                $mimeTypeGuesser->isSupported() ? $mimeTypeGuesser->guess($file) : 'image/jpeg'
+            );
+
+            $response->setContentDisposition(
+                ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+                $customizationForViewing->getValue()
+            );
+
+            return $response;
```

Comment:
Why not using `$this->file` instead? https://symfony.com/blog/new-in-symfony-3-2-file-controller-helper
Let Symfony do the job :wink:

Reply:
Didn't know about this, it's much cleaner indeed 👍